resources/migrations/001_update_migrations.yaml

```diff
@@ -9270,6 +9270,90 @@ databaseChangeLog:
                   type: ${text.type}
                   remarks: running, failed, or completed
 
+
+  - changeSet:
+      id: v51.2024-08-28T07:03:41
+      author: johnswanson
+      comment: Add `report_card.dashboard_id`
+      changes:
+        - addColumn:
+            tableName: report_card
+            columns:
+              - column:
+                  name: dashboard_id
+                  type: int
+                  remarks: The dashboard that owns the card, if it is a dashboard-internal card.
+                  constraints:
+                    nullable: true
+
+  - changeSet:
+      id: v51.2024-08-28T07:03:42
+      author: johnswanson
+      comment: Add `report_dashboardcard.is_dashboard_internal_card`
+      changes:
+        - addColumn:
+            tableName: report_dashboardcard
+            columns:
+              - column:
+                  name: is_dashboard_internal_card
+                  type: ${boolean.type}
+                  remarks: |
+                    A denormalized representation that this edge from dashboard<->card should be the *only* edge from its card.
+                    Provides an efficient way to enforce the fact that
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
+
+  - changeSet:
+      id: v51.2024-08-28T07:03:43
+      author: johnswanson
+      comment: Add `report_dashboardcard.dashboard_internal_card_id`
```

Comment:
This unique field won't guarantee that there aren't some non-`is_dashboard_internal_card` dashcards for the given card 🤔

Reply:
Ah, if I understand the product right, this is not an issue?

If we move a "dashboard card" to a collection, would we leave the `dashboard_internal` fields in place?

Reply:
See above, I [ended up removing this column entirely](https://github.com/metabase/metabase/pull/47369#discussion_r1759449149).